state/keys_test.go

```diff
@@ -84,3 +84,116 @@ func TestMalformedKey(t *testing.T) {
 	keys := make(Keys)
 	require.False(keys.Add("", Read))
 }
+
+func TestHasPermissions(t *testing.T) {
+	tests := []struct {
+		name     string
+		perm     Permissions
+		contains Permissions
+		failing  bool
+	}{
+		{
+			name:     "read is read",
```

Comment:
Let's change the names from "X is X" and "X not Y" -> "X has Y" or "X does not have Y"

Reply:
Done, thanks!